pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -921,17 +929,20 @@ func (o *operatingSystemConfig) calculateKeyForVersion(version int, worker *gard
 	if err != nil {
 		return "", err
 	}
+	kubeReserved := v1beta1helper.CalcluateEffectiveResourceReservations(o.values.KubeletConfig, worker.Kubernetes)
 
-	return CalculateKeyForVersion(version, kubernetesVersion, worker)
+	return CalculateKeyForVersion(version, kubernetesVersion, o.values.CredentialsRotationStatus, worker, o.values.NodeLocalDNSEnabled, kubeReserved)
```

Comment:
Nit: Why not passing `o.values` to make it easier to extend this in the future?

Reply:
Is changed.